elastic/kibana
# [Index Management] Add data streams functionality to indices tab

## Summary

- Migrate the URL to be the source of truth for the includeHiddenIndices toggle
- Add a new url param dataStreams that filters out indices based on the data stream they are backing

### How to test

1. Create a data stream in your ES cluster: `PUT _data_stream/my-data-stream`
2. Go to Index Management plugin in Kibana (requires x-pack). There should be no indices visible if this is a fresh cluster.
3. Press the include hidden toggle and all hidden indices should appear including `my-data-stream-000001`
4. You should be able to navigate to the url you are on with `&filter=dataStream:my-data-stream` added to the hash portion of your url. This filters for just the data stream backing index `my-data-stream-000001`. Something like: `/app/kibana#/management/data/index_management/indices?includeHiddenIndices=true&filter=dataStream:my-data-stream`

### Notes to reviewer

* A refactor was done to remove the middleware for syncing the URL to the store. State was moved down to the index table where it is derived from the location object we receive from `withRouter`.
* The selector `getFilteredIndices` still does the work of filtering over all indices but now also depends on the component props, specifically the location.search value to determine what all it should be filtering. More information on this [here](https://github.com/reduxjs/reselect#accessing-react-props-in-selectors).
* We can do the same for `filter` as we did for `includeHiddenIndices`; move the source of truth to the URL. This will be a slightly bigger refactor because of how it is currently implemented and can be set by other plugins (like ILM). The filter will need to be converted to a string value and parsed into `Query` by us. I would consider this follow up work, but it would be nice to have the actual text in the filter sync up with the URL as the user types.

### Screenshots

**Filtering based on hidden**
<img width="1133" alt="Screenshot 2020-06-02 at 14 23 17" src="https://user-images.githubusercontent.com/8155004/83519636-a36fe080-a4dc-11ea-9e23-339c7839ac59.png">

**Filtering based on text filter and include hidden**
<img width="1603" alt="Screenshot 2020-06-02 at 20 52 58" src="https://user-images.githubusercontent.com/8155004/83558306-1dbb5780-a513-11ea-8cf9-18be9b04f473.png">


### Checklist


### For maintainers
